### PR TITLE
Extract ifdefed functions

### PR DESCRIPTION
**Pull request by tupaschoal**
Easier to follow per commit. I'm just moving stuff that branches on platform to the actual platform

### CODE REVIEW

File: src/openrct2/core/FileScanner.cpp [hunk 1]
```diff
@@ -23,335 +23,156 @@
...
+}
 
 IFileScanner* Path::ScanDirectory(const std::string& pattern, bool recurse)
 {
-#ifdef _WIN32
-    return new FileScannerWindows(pattern, recurse);
-#elif defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
-    return new FileScannerUnix(pattern, recurse);
-#endif
+    return Platform::ScanDirectory(pattern, recurse);
```
Comment (duncanspumpkin) [inline src/openrct2/core/FileScanner.cpp:175]: Add a Todo saying replace with std::unique_ptr and create an issue to track it.

Comment (tupaschoal) [reply src/openrct2/core/FileScanner.cpp:176]: Done

**Comment by Gymnasiast** [comment]
I'd prefer if the string functions were more clearly labelled as such: `StringToUpper`, etc.
Also, the functions `ToWidechar` and `ToUTF8` lack some context and should be called `UTF8ToWideChar` and `WidecharToUTF8`. That is, if they even _belong_ here - they're only used on Windows, as macOS/Linux/BSD/Android all use UTF-8 directly.

**Comment by IntelOrca** [comment]
> That is, if they even belong here - they're only used on Windows, as macOS/Linux/BSD/Android all use UTF-8 directly.

It might have been OpenLoco, but I think sometimes widechar is used for some APIs. Their widechar differs in that it is 32-bit rather than 16-bit - but the same named function can still be used in those instances.

**Comment by tupaschoal** [comment]
Took another route first, but could move some stuff out to the Platform, finally.

File: src/openrct2/core/Path.cpp [hunk 2]
```diff
@@ -202,42 +191,15 @@ namespace Path
         return lastDot;
     }
 
-    utf8* GetAbsolute(utf8* buffer, size_t bufferSize, const utf8* relativePath)
+    utf8* GetAbsolutePath(utf8* buffer, size_t bufferSize, const utf8* relativePath)
     {
```
Comment (duncanspumpkin) [inline src/openrct2/core/Path.cpp:195]: I don't agree with this rename. We are already in the path namespace. Also you've only renamed one of the overloads

Comment (tupaschoal) [reply src/openrct2/core/Path.cpp:195]: I thought I had dropped this renaming and left the new name only on Platform, but seems like I hadn't. Just rewrote it.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**